x/swap/keeper/deposit.go

```diff
@@ -75,6 +76,15 @@ func (k Keeper) Deposit(ctx sdk.Context, depositor sdk.AccAddress, coinA sdk.Coi
 		return sdkerrors.Wrapf(types.ErrSlippageExceeded, "slippage %s > limit %s", slippage, slippageLimit)
 	}
 
+	k.updatePool(ctx, poolID, pool)
+	if shareRecord, hasExistingShares := k.GetDepositorShares(ctx, depositor, poolID); hasExistingShares {
+		k.BeforePoolDepositModified(ctx, poolID, depositor, shareRecord.SharesOwned)
+		k.updateShares(ctx, depositor, poolID, shareRecord.SharesOwned.Add(shares))
+	} else {
+		k.AfterPoolDepositCreated(ctx, poolID, depositor, shares)
+		k.updateShares(ctx, depositor, poolID, shares)
```

Comment:
nit: I'd maybe switch the order of these just to keep the `After...` name accurate. Then if any hook listener calls `GetDepositorShares`, the new shares will be in the store.

Reply:
Great catch! Wasn't thinking about ordering.  I added a regression test to cover all hook ordering, and fixed